state/app/DataManager.kt

```diff
@@ -93,9 +93,33 @@ class DataManager {
             set(value) = value?.let { setProperty(CONNECTION_CA_CERTIFICATE, it) } ?: Unit
     }
 
+    inner class Preferences {
+        private val AUTO_SAVE_DEFAULT = "editor.autosave"
+        private val IGNORED_PATH_DEFAULT = "project.ignoredpaths"
+        private val QUERY_LIMIT = "query.limit"
+        private val GRAPH_OUTPUT = "graph.output"
+
+        var autoSave: Boolean
+            get() = properties?.getProperty(AUTO_SAVE_DEFAULT)?.toBoolean() ?: true
+            set(value) = setProperty(AUTO_SAVE_DEFAULT, value.toString())
+
+        var ignoredPaths: List<String>
+            get() = properties?.getProperty(IGNORED_PATH_DEFAULT)?.split(',') ?: ".git".split(',')
+            set(value) = setProperty(IGNORED_PATH_DEFAULT, value.joinToString())
+
+        var limit: String
+            get() = properties?.getProperty(QUERY_LIMIT) ?: "1000"
+            set(value) = setProperty(QUERY_LIMIT, value)
+
+        var graphOutput: Boolean
+            get() = properties?.getProperty(GRAPH_OUTPUT)?.toBoolean() ?: true
+            set(value) = setProperty(GRAPH_OUTPUT, value.toString())
```

Comment:
Properties aren't created at first boot, and we can't guarantee that the user will set their preferences before running their first query or opening their first file. As such, we guarantee that these values will return something valid by supplying default values in the event of a null return from `getProperty`.

Reply:
This has been implemented. We've brought `limit` up to speed too, calling it `matchQueryLimit`.

`PreferenceManager` looks much better now, it provides the rest of the application data in the format it will want it and hides the implementation details of storing preferences as strings underneath.